src/object/player.cpp

```diff
@@ -360,6 +367,58 @@ Player::update(float dt_sec)
 
   // extend/shrink tux collision rectangle so that we fall through/walk over 1
   // tile holes
+  
+  //wallclinging and walljumping
+  
+  Rectf wallclingleft = get_bbox();
+  wallclingleft.set_left(wallclingleft.get_left() - 8);
+  bool trygrabwallleft = !Sector::get().is_free_of_statics(wallclingleft);
+  m_on_left_wall = ((trygrabwallleft) ? true : false);
+
+  Rectf wallclingright = get_bbox();
+  wallclingright.set_right(wallclingright.get_right() + 8);
+  bool trygrabwallright = !Sector::get().is_free_of_statics(wallclingright);
+  m_on_right_wall = ((trygrabwallright) ? true : false);
+  
+  if (m_on_wall)
+  {
+    if (((trygrabwallleft) || (trygrabwallright)) && !on_ground() && !m_does_buttjump && !m_swimming)
+    {
+      m_on_wall = true;
+    }
+    else
+    {
+      m_on_wall = false;
+    }
+  }
+  else
+  {
+    if (((trygrabwallleft) || (trygrabwallright)) && !on_ground() && !m_does_buttjump && !m_swimming)
+    {
+      m_on_wall = true;
+    }
```

Comment:
When entering this point of the code, `m_on_wall` is guaranteed to be `false`, so that code does the same job as the code in the part above it. Replace the whole `if (m_on_wall) { ... } else { ... }` with the code I gave you in the other review